File: 0152-maximum-product-subarray/0152-maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {
        int pre=1;
        int suff=1;
        int n=nums.length;

        int maxi=Integer.MIN_VALUE;

        for(int i=0;i<nums.length;i++){
            if(pre==0)pre=1;
            if(suff==0)suff=1;

            pre=pre*nums[i];
            suff=suff*nums[n-i-1];

            maxi=Math.max(maxi,Math.max(pre,suff));
        }

        return maxi;
    }
}

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int pre=1;
        int suf=1;
        int n=nums.size();

        int maxi=INT_MIN;

        for(int i=0;i<nums.size();i++){
            if(pre==0)pre=1;
            if(suf==0)suf=1;

            pre=pre*nums[i];
            suf=suf*nums[n-i-1];
            maxi=max(maxi,max(pre,suf));

        }

        return maxi;
    }
};